src/core/PathUtil.cpp

```diff
@@ -0,0 +1,139 @@
+#include "PathUtil.h"
+
+#include <QDebug>
+#include <QDir>
+#include <QFileInfo>
+
+#include "ConfigManager.h"
+
+namespace PathUtil
+{
+	Base relativeBases[] = { ProjectDirBase, FactorySampleBase, UserSampleBase, UserVSTBase, PresetBase,
+		UserLADSPABase, DefaultLADSPABase, UserSoundfontBase, DefaultSoundfontBase, UserGIGBase, DefaultGIGBase };
+
+	QString baseLocation(const Base & base)
+	{
+		QString loc = "";
+		switch (base)
+		{
+			case ProjectDirBase       : loc = ConfigManager::inst()->userProjectsDir(); break;
+			case FactorySampleBase    :
+			{
+				QDir fsd = QDir(ConfigManager::inst()->factorySamplesDir());
+				loc = fsd.absolutePath() + "/"; break;
+			}
+			case UserSampleBase       : loc = ConfigManager::inst()->userSamplesDir(); break;
+			case UserVSTBase          : loc = ConfigManager::inst()->userVstDir(); break;
+			case PresetBase           : loc = ConfigManager::inst()->userPresetsDir(); break;
+			case UserLADSPABase       : loc = ConfigManager::inst()->ladspaDir(); break;
+			case DefaultLADSPABase    : loc = ConfigManager::inst()->userLadspaDir(); break;
+			case UserSoundfontBase    : loc = ConfigManager::inst()->sf2Dir(); break;
+			case DefaultSoundfontBase : loc = ConfigManager::inst()->userSf2Dir(); break;
+			case UserGIGBase          : loc = ConfigManager::inst()->gigDir(); break;
+			case DefaultGIGBase       : loc = ConfigManager::inst()->userGigDir(); break;
+			default                   : return QString("");
+		}
+		return QDir::cleanPath(loc) + QDir::separator();
```

Comment:
`QDir::cleanPath` converts native separators to "/", but `QDir::separator` returns a native separator. The [documentation for `QDir::separator`](https://doc.qt.io/qt-5/qdir.html#separator) says
> You do not need to use this function to build file paths. If you always use "/", Qt will translate your paths to conform to the underlying operating system.

and Windows is happy to accept forward instead of backward slashes anyway if there's somewhere we don't use Qt to access files. So perhaps just use "/"?